### ISSUE TITLE
flipped output when translating to arabic 

### ISSUE THREAD

**Issue by hamadaspace** [soimort/translate-shell#288]
**Version:**

```
[0]:~$ trans -V
Translate Shell       0.9.6.9-git:eba74aa
platform              Linux
gawk (GNU Awk)        4.1.4
fribidi (GNU FriBidi) 0.19.7
audio player          mpv --no-config
terminal pager        less
terminal type         xterm-256color
user locale           en_US.UTF-8 (English)
home language         en
source language       auto
target language       en
translation engine    google
proxy                 [NONE]
user-agent            Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/602.1 (KHTML, like Gecko) Version/8.0 Safari/602.1 Epiphany/3.18.2
theme                 default
init file             [NONE]

Report bugs to:       https://github.com/soimort/translate-shell/issues

```
NOTE: trans was installed with apt ! (translate-shell - Google Translate to serve as a command line tool)

**Commands:**

```
[0]:~$ trans -b :ar "Hello world"
                                                                                                                                            ﻢﻟﺎﻌﻟﺎﺑ ﺎﺒﺣﺮﻣ
[0]:~$ trans -b fr:ar "Bonjour le monde"
                                                                                                                                             ﻢﻟﺎﻌﻟﺍ ﺎﺒﺣﺮﻣ
[0]:~$ trans -b :ar "Hello world"|rev
ﻣﺮﺣﺒﺎ ﺑﺎﻟﻌﺎﻟﻢ                                                                                                                                            
[0]:~$ trans -b fr:ar "bonjour le monde"|rev
ﻣﺮﺣﺒﺎ ﺍﻟﻌﺎﻟﻢ                                                                                                                                             

```
**Additional infos:**

_as you see i can handle that with "rev" command but it works only with "Brief mode" !_

```
[0]:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.2 LTS
Release:        18.04
Codename:       bionic
```

**Comment by soimort** [comment]
`trans` should be able to call `fribidi` to handle Arabic scripts -- unless `fribidi` is doing it wrong. It works on my machine:

![Screenshot from 2019-03-11 19-34-23](https://user-images.githubusercontent.com/342945/54149805-4f8c5e00-4437-11e9-8663-c3cefa0dfe3b.png)

Can you confirm the results of these? (use screenshots)
```
$ trans -no-ansi -no-bidi -b :ar "Hello world"
$ trans -no-ansi -no-bidi -b :ar "Hello world" | fribidi
```
![Screenshot from 2019-03-11 19-53-22](https://user-images.githubusercontent.com/342945/54150180-1ef8f400-4438-11e9-8536-7e67e8dcfdf8.png)

**Comment by jonashamada** [comment]
to be quick : i couldn't add screenshot !

`
[0]:~$ trans -no-ansi -no-bidi -b :ar "Hello world"
مرحبا بالعالم
[0]:~$ trans -no-ansi -no-bidi -b :ar "Hello world" | fribidi
                                                                   ﻢﻟﺎﻌﻟﺎﺑ ﺎﺒﺣﺮﻣ
[0]:~$ trans -no-bidi -b :ar "Hello world"
مرحبا بالعالم
[0]:~$ `

_It's working with '-no-bidi'_

**Comment by hamadaspace** [comment]
thanks a lot , it's working with '-no-bidi' !

![out](https://user-images.githubusercontent.com/7035529/54152186-8b71f400-4434-11e9-9ba3-8bac4345c256.png)

for me i've no issue now but for the completness of the program !
![out](https://user-images.githubusercontent.com/7035529/54152878-2cad7a00-4436-11e9-89be-ab6acf50ea72.png)

the selected word is a flipped one it shold be 'ﺍﻟﻌﺮﺑﻴﺔ'

**Comment by soimort** [comment]
What is your terminal emulator?

VTE-based terminals such as `gnome-terminal` and `sakura` should work just fine with `trans`. However, if you're on a terminal that implements its own multilingual support such as `mlterm`, it can mess up with the RTL support of `trans` -- that is, while `trans`already converted an Arabic string to its right-to-left form for display (via `fribidi`), `mlterm` blindly inverts it again, then you get the original left-to-right Arabic string, which is wrong.

I can reproduce the same issue on `mlterm`, so I guess that might be what you're using.

**Comment by hamadaspace** [comment]
my terminal is Yakuake which uses Konsole (kde terminal), and yes you are right , i tried another terminal that comes with xfce4 and it's just working perfectly!
![out](https://user-images.githubusercontent.com/7035529/54162508-95ecb780-444d-11e9-9dcf-47ac3269a500.png)

**Comment by soimort** [comment]
Should be fixed by 3bfcebb (and a bunch of other relevant commits) now. According to [this thread](https://bugs.launchpad.net/ubuntu/+source/vte/+bug/263822), there are mainly two terminal emulators that implemented their own Bi-Di text rendering:
1. `konsole` and its derivatives (such as `yakuake`).
2. `mlterm`

`trans` will no longer invoke Fribidi on those, leaving the right-to-left rendering task to the emulator.

![Screenshot from 2019-03-12 23-26-24](https://user-images.githubusercontent.com/342945/54240982-0d8b1700-4520-11e9-90df-da40f3dcf5ad.png)

The solution comes with two limitations though:
1. It won't work inside an SSH session. If you connect to a remote server and use translate-shell from there, it has no way to tell which terminal emulator you're running locally.
2. On `konsole` and `yakuake`, Bi-Di rendering can be disabled by users if desired. Since `trans` now assumes you have that feature on, turning it off will cause display issues again. In that case, the option `-bidi` can be used to enforce the usage of Fribidi.

![Screenshot from 2019-03-12 23-25-37_](https://user-images.githubusercontent.com/342945/54241428-2ba54700-4521-11e9-98dc-a6101e69d50b.png)
